src/utils/restrictToRole.js

```diff
@@ -0,0 +1,20 @@
+const { Forbidden, BadRequest } = require('../errors');
+const { userRepo } = require('../components/user/repo/index');
+
+const restrictToRole = async (id, roleName, app) => {
+	if (!id) {
+		throw new BadRequest('The request query should include a valid userId');
+	}
+	if (!roleName) {
+		throw new BadRequest('The request query should include a valid roleName');
+	}
+
+	const roles = await userRepo.getUserRoles(id, app);
```

Comment:
Restricted to role? What we want is restricted to permissions, or not?

Reply:
Maybe the naming is not good. What we want to achieve here is to ensure the deleted object matches to the permission, i.e.
STUDENT_DELETE = Delete only student
TEACHER_DELETE = Delete only teacher